docs/readthedocs/source/doc/error-log-api.md

```diff
@@ -0,0 +1,109 @@
+BigDL provides error handling api. Please don't use `assert`, `raise`, `throw` to fail the application.
+Use the error handling api instead, it will provide useful message for debugging.
+
+## **Error handling API**
+
+**Scala**
+
+Import package
+```scala
+import com.intel.analytics.bigdl.dllib.utils.Log4Error
+```
+
+If user's input is invalid
+```scala
+Log4Error.invalidInputError(condition: Boolean, errmsg: String, fixmsg: String = null)
+```
+
+* `condition`: Will throw exception and print errmsg if it's `true`. Otherwise will pass the function
+* `errmsg`: Error message to be print.
+* `fixmsg`: Message about how to fix the error.
+
+If user call the api wrong
+```scala
+Log4Error.invalidOperationError(condition: Boolean, errmsg: String, fixmsg: String = null, cause: Throwable = null)
+```
+
+* `condition`: Will throw exception and print errmsg if it's `true`. Otherwise will pass the function
+* `errmsg`: Error message to be print.
+* `fixmsg`: Message about how to fix the error.
+* `cause`: Exception need to throw.
+
+For unkown Exception:
+```scala
+Log4Error.unKnowExceptionError(condition: Boolean, errmsg: String, fixmsg: String = null, cause: Throwable = null)
+```
+
+* `condition`: Will throw exception and print errmsg if it's `true`. Otherwise will pass the function
+* `errmsg`: Error message to be print.
+* `fixmsg`: Message about how to fix the error.
+* `cause`: Exception need to throw.
+
+Notes: This API is for future extension, in case we need distinct invalidOperation exception with unKnownException in python
+
+
+**Python**
+
+Import package
+```python
+from bigdl.dllib.utils.log4Error import *
+```
+
+If user's input is invalid
+```python
+invalidInputError(condition, errMsg, fixMsg=None)
+```
+
+* `condition`: Will throw exception and print errmsg if it's `true`. Otherwise will pass the function
```

Comment:
Actually, the behavior is "Will throw exception and print errmsg if it's `false`"
```
def invalidInputError(condition, errMsg, fixMsg=None):
    if not condition:
        outputUserMessage(errMsg, fixMsg)
        raise RuntimeError(errMsg)
```

Reply:
Thanks for pointing out. Fixed